segmatch/include/segmatch/common.hpp

```diff
@@ -131,10 +132,46 @@ struct IdIndex {
   size_t index = 0u;
 };
 
+static std::string getFileExtension(const std::string& filename) {
+  auto str_iter = filename.rbegin();
+  for (; str_iter != filename.rend(); ++str_iter) {
+    if (*str_iter == '.') break;
+  }
+  return std::string(str_iter.base(), filename.rbegin().base());
```

Comment:
What's the proper way to do this?
`filename.rbegin().base()` or `filename.end()`.
Both are equivalent.